network/eni/eni.go

```diff
@@ -102,6 +97,27 @@ func (eni *ENI) AttachToLink() error {
 	return nil
 }
 
+// getInterfaceByMac gets the interface with specified mac address, and picks the one with shortest name
```

Comment:
Let's not discuss the trunk vs. branch MAC address here. The ENI class doesn't know about trunks etc.
// getInterfaceByMACAddress returns the interface with the specified MAC address.

Move the rest of your comment block to just before the for loop (line 106):
// If there are multiple matches, pick the one with the shortest name.

Reply:
updated